Sources/MessagesViewController.swift

```diff
@@ -1,28 +1,146 @@
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE. 
+ */
 
-import UIKit
-import Foundation
 
+import UIKit
 
-public final class MessagesViewController: UICollectionViewController {
+open class MessagesViewController: UIViewController {
+    
+    // MARK: - Properties
+    
+    open var messagesCollectionView: MessagesCollectionView = {
+        let flowLayout = MessagesCollectionViewFlowLayout()
+        let messagesCollectionView = MessagesCollectionView(frame: .zero, collectionViewLayout: flowLayout)
+        messagesCollectionView.backgroundColor = .gray // color for testing
+        return messagesCollectionView
+    }()
+    
+    open var messageInputBar: MessageInputBar = {
+        
+        let messageInputBar = MessageInputBar(frame: .zero)
+        messageInputBar.backgroundColor = .lightGray // color for testing
+        return messageInputBar
+    }()
+    
+    // MARK: - View Life Cycle
 
-    override public func viewDidLoad() {
+    override open func viewDidLoad() {
         super.viewDidLoad()
+        
+        tabBarController?.tabBar.isHidden = true
+        
+        setupSubviews()
+        setupConstraints()
+        
+        messagesCollectionView.register(MessageCollectionViewCell.self, forCellWithReuseIdentifier: "cell")
+        
+        messagesCollectionView.delegate = self
+        messagesCollectionView.dataSource = self
+        
+    }
+    
+    override open func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        tabBarController?.tabBar.isHidden = false
+        
+    }
+    
+    // MARK: - Methods
+    
+    func setupSubviews() {
+        view.addSubview(messagesCollectionView)
+        view.addSubview(messageInputBar)
+    }
+    
+    func setupConstraints() {
+
+        messagesCollectionView.translatesAutoresizingMaskIntoConstraints = false
+        view.addConstraint(NSLayoutConstraint(item: messagesCollectionView, attribute: .top, relatedBy: .equal, toItem: topLayoutGuide, attribute: .bottom, multiplier: 1, constant: 0))
```

Comment:
Yes, minimum targets. I struggled with the decision to support iOS 8. I figured why not unless it becomes a hassle.